lib/cppcheck.cpp

```diff
@@ -686,6 +689,11 @@ unsigned int CppCheck::checkClang(const FileWithDetails &file, int fileIndex)
         mErrorLogger.reportOut(exe + " " + args2, Color::Reset);
     }
 
+    if (!mExecuteCommand) {
+        std::cerr << "Failed to execute '" << exe << " " << args2 << " " << redirect2 << "' - (no command callback provided)" << std::endl;
```

Comment:
why isn't this reported to the errorlogger. GUI does not capture stderr. And plugins will probably not write this. So the user will have no clue why it does not work.

Reply:
I don't understand what I am missing. What is stopping you from calling `mErrorLogger.reportError` here instead?

Reply:
If we write this code instead:
```
        const std::list<ErrorMessage::FileLocation> callstack{
            {file.spath(), -1, 0}
        };
        const ErrorMessage msg(callstack, nullptr, Severity::error, "internalError", "Failed to execute clang, mExecuteCommand is null", Certainty::normal);
        mErrorLogger.reportErr(msg);
```
It still compiles. I don't ask that you change other std::cerr/std::cout statements throughout the code, those should be fixed but I agree that is out of scope.